packages/framework-provider-aws/src/library/searcher-adapter.ts

```diff
@@ -32,62 +31,150 @@ export async function searchReadModel(
-      return `NOT contains(#${propName}, ${holder(0)})`
-    case 'begins-with':
-      return `begins_with(#${propName}, ${holder(0)})`
-    default:
-      throw new InvalidParameterError(`Operator "${filter.operation}" is not supported`)
-  }
+  return Object.entries(filter)
+    .map(([operation, value], index) => {
+      switch (operation) {
+        case 'eq':
+          return `#${propName} = ${holder(index)}`
+        case 'ne':
+          return `#${propName} <> ${holder(index)}`
+        case 'lt':
+          return `#${propName} < ${holder(index)}`
+        case 'gt':
+          return `#${propName} > ${holder(index)}`
+        case 'gte':
+          return `#${propName} >= ${holder(index)}`
+        case 'lte':
+          return `#${propName} <= ${holder(index)}`
+        case 'in':
+          return `#${propName} IN (${value
+            .map((value: any, subIndex: number) => `${holder(index, subIndex)}`)
+            .join(',')})`
+        case 'contains':
+          return `contains(#${propName}, ${holder(index)})`
+        case 'beginsWith':
+          return `begins_with(#${propName}, ${holder(index)})`
+        case 'includes':
+          return `contains(#${propName}, ${holder(index)})`
+        default:
+          if (typeof value === 'object') {
+            return `#${propName}.${buildOperation(operation, value)}`
+          }
+          throw new InvalidParameterError(`Operator "${operation}" is not supported`)
+      }
+    })
+    .join(' AND ')
 }
 
-function placeholderBuilderFor(propName: string): (valueIndex: number) => string {
-  return (valueIndex: number) => `:${propName}_${valueIndex}`
+const attrCount: string[] = []
+function resetAttrCount(): void {
+  attrCount.length = 0
+}
+function placeholderBuilderFor(propName: string): (valueIndex: number, valueSubIndex?: number) => string {
+  return (valueIndex: number, valueSubIndex?: number) => {
+    const placeholder = `:${propName}_${valueIndex}` + (typeof valueSubIndex === 'number' ? `_${valueSubIndex}` : '')
+    if (attrCount.includes(placeholder)) return placeholderBuilderFor(propName)(valueIndex + 1)
+    attrCount.push(placeholder)
+    return placeholder
+  }
 }
 
-function buildExpressionAttributeNames(filters: Record<string, Filter<any>>): ExpressionAttributeNameMap {
+function buildExpressionAttributeNames(filters: FilterFor<any>): ExpressionAttributeNameMap {
   const attributeNames: ExpressionAttributeNameMap = {}
   for (const propName in filters) {
-    attributeNames[`#${propName}`] = propName
+    switch (propName) {
+      case 'not':
+        Object.assign(attributeNames, buildExpressionAttributeNames(filters[propName] as FilterFor<any>))
+        break
+      case 'and':
+      case 'or':
+        for (const filter of filters[propName] as Array<FilterFor<any>>) {
+          Object.assign(attributeNames, buildExpressionAttributeNames(filter as FilterFor<any>))
+        }
+        break
+      case 'includes':
+        break
```

Comment:
What happens with this case? If we're planning to implement it in the future it could make sense to leave a TODO here:

```suggestion
      case 'includes':
        // TODO: To be implemented in the future. More details in issue #XYZ
        break
```

Reply:
I _think_ this is OK, because, after the include operation, there is only a value. There are no attribute names that needs to be accounted for in the AWS query.

Anyway, if that is the case, I agree that it would be great to have a comment explaining that.

Reply:
This is because when there is an includes operation it shouldn't enter in the `default` case but get ingnored instead.

Reply:
Then I'd add a comment explaining that or someone could end up removing that code thinking that's useless..